src/generators/cplusplus/CplusplusConstrainer.ts

```diff
@@ -0,0 +1,110 @@
+import { ConstrainedObjectPropertyModel } from '../../models';
+import { TypeMapping } from '../../helpers';
+import {
+  defaultEnumKeyConstraints,
+  defaultEnumValueConstraints
+} from './constrainer/EnumConstrainer';
+import { defaultModelNameConstraints } from './constrainer/ModelNameConstrainer';
+import { defaultPropertyKeyConstraints } from './constrainer/PropertyKeyConstrainer';
+import { CplusplusDependencyManager } from './CplusplusDependencyManager';
+import { CplusplusOptions } from './CplusplusGenerator';
+
+function ensureOptional(
+  type: string,
+  partOfProperty: ConstrainedObjectPropertyModel | undefined,
+  dependencyManager: CplusplusDependencyManager
+): string {
+  if (partOfProperty !== undefined) {
+    dependencyManager.addDependency('#include <optional>');
+    return `std::optional<${type}>`;
+  }
+  return type;
+}
```

Comment:
This doesn't seem to do what I expect it to do. Even if I mark an object's property as required, it still ends up being generated as an optional field.

Reply:
A bit too fast with the implementation 😅 Was missing `&& partOfProperty.required !== true`.

✅